airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -1261,14 +1265,36 @@ def add_to_outputs(
                 else:
                     hooks.append(f"drop view {stg_schema}.{stg_table}")
 
-                # Explicit function so that we can have type hints to satisfy the linter
-                def wrap_in_quotes(s: str) -> str:
-                    return '"' + s + '"'
-
                 config["post_hook"] = "[" + ",".join(map(wrap_in_quotes, hooks)) + "]"
             else:
                 # incremental is handled in the SCD SQL already
                 sql = self.add_incremental_clause(sql)
+        elif self.destination_sync_mode == DestinationSyncMode.overwrite:
+            if suffix == "" and not is_intermediate:
+                # drop SCD table after creating the destination table
+                scd_table_name = self.tables_registry.get_table_name(schema, self.json_path, self.stream_name, "scd", truncate_name)
+                print(f"  Adding drop table hook for {scd_table_name} to {file_name}")
+                hooks = [
+                    Template(
+                        """
+                    {{ '{%' }}
+                        set scd_table_relation = adapter.get_relation(
+                            database=this.database,
+                            schema=this.schema,
+                            identifier='{{ scd_table_name }}'
+                        )
+                    {{ '%}' }}
+                    {{ '{%' }}
+                        if scd_table_relation is not none
+                    {{ '%}' }}
+                    {{ '{%' }}
+                            do adapter.drop_relation(scd_table_relation)
+                    {{ '%}' }}
+                    {{ '{% endif %}' }}
```

Comment:
does this work if you run an incremental dedup sync, then switch the stream to full refresh overwrite and do a reset? I suspect it'll fail because the query is empty

(probably just need to do an `else: select 0` to make the destination DB/DW happy)

Reply:
er, actually you would just need to set up a stream in full-refresh, no need for the incremental thing. I.e.:
1. full-refresh overwrite sync, starting from an empty DB
2. this hook will execute, and attempt to execute `""` as a SQL query

Reply:
I think, this is exactly what my integration test does:
 - do a full-refresh overwrite sync to get rid of anything left over from previous runs
 - sync in incremental dedup mode
 - do a full refresh overwrite with new schema
 - sync in incremental dedup mode with new schema

Reply:
actually, from reading through the test - I think it's not running normalization after the initial reset? I.e. the test is doing:
1. wipe `_raw` tables (i.e. `write` with `sync_mode: overwrite` + 0 records)
2. write data into `_raw` tables (`write` with `sync_mode: append dedup` + nonzero records)
3. (everything else)

my theory is that if you try to run normalization [here](https://github.com/airbytehq/airbyte/pull/18015/files#diff-971224e0addaf3252ea20188de0265da4c40dd532702ef9122c8bece648e94d2R99) that it will throw an error

Reply:
Good find. I will look into this

Reply:
I added a normalization run after initial reset (where there should be no tables at all) and things still work fine. At least in BQ.

Reply:
nice! :shipit:

Reply:
I also tested this manually - reset works fine, because Airbyte recreates empty _raw and destination tables after a reset.

Reply:
I also tested a reset with empty target schema and empty source table - we still create target table and _raw table and normalization does not try to run an empty query